proposals/0184-block-writeable-account-data-limit.md

```diff
@@ -0,0 +1,107 @@
+---
+simd: '0184'
+title: Block Writeable Account Data Limit
+authors:
+  - Brennan Watt (Anza)
+category: Standard
+type: Core
+status: Idea
+created: 2024-10-15
+feature: (fill in with feature tracking issues once accepted)
+supersedes:
+superseded-by:
+extends:
+---
+
+## Summary
+
+A protocol limit on the amount of account data in bytes that may be written
+during a single block.
+
+## Motivation
+
+When direct mapping is enabled, clients will be able to handle much more account
+data because there will be fewer copies in memory (e.g. for each VM invocation /
+CPI call). However, writeable account data cannot be direct mapped because it
+needs to be written back after a successful tx execution. This necessitates a
+decoupling between account read and write data in order to increase readable
+account data limits.
+
+Note: There is an existing per block limit on newly allocated account data
+(100MB), but this does not encompass account data being updated/written back.
+
+## New Terminology
+
+None
+
+## Detailed Design
+
+This feature would check the amount of account data marked as writeable for each
+transaction being handled by the block producer (for inclusion in a block) or
+validator (replaying transactions in a block), add it to a running total of
+total writeable account data for the block, and compare it against some
+threshold (recommend 2GB to start) to decide if the transaction is allowed to be
```

Comment:
Do we have statistics on what is the current distribution of writable data on the mainnet?
I wonder if we could start with an even lower value.

Reply:
We don't have perfect stats, but I posted comment with some analysis below (https://github.com/solana-foundation/solana-improvement-documents/pull/184#issuecomment-2432814006)

2GiB feels right as a starting point. It should constrain worst case reasonably well without impacting current typical behavior on mainnet. We could ratchet things down (or up) after observing behavior